src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/dllmain.cpp

```diff
@@ -29,12 +68,12 @@ VOID CALLBACK EnsureMTAInitializedCallBack
     CoGetObjectContext(IID_PPV_ARGS(&spThreadingInfo));
 }
 
-/*
-In the context callback call to the MTA apartment, there is a bug that prevents COM
-from automatically initializing MTA remoting. It only allows NTA to be intialized
-outside of the NTA and blocks all others. The workaround for this is to spin up another
-thread that is not been CoInitialize. COM treats this thread as a implicit MTA and
-when we call CoGetObjectContext on it we implicitily initialized the MTA.
+/* 
+In the context callback call to the MTA apartment, there is a bug that prevents COM 
+from automatically initializing MTA remoting. It only allows NTA to be intialized 
+outside of the NTA and blocks all others. The workaround for this is to spin up another 
+thread that is not been CoInitialize. COM treats this thread as a implicit MTA and 
+when we call CoGetObjectContext on it we implicitily initialized the MTA. 
 */
 HRESULT EnsureMTAInitialized()
```

Comment:
Wouldn't it be simpler to use CoIncrementMTAUsage?

Reply:
No idea.  @kythant any ideas?

Reply:
CoIncrementMTAUsage is what C++/WinRT uses to ensure the MTA is present. This function was introduced in Windows 8, so no good if you're targeting Windows 7 but it looks like you're calling all kinds of other Windows 8 functions as well.

Reply:
COM bug in CoIncrementMTAUsage is preventing it from initializing MTA remoting and the callback below was failing with RPC_E_DISCONNECTED error. Thus the reason for this implicit MTA hack.